packages/proxy/lib/http/response-middleware.ts

```diff
@@ -773,7 +773,24 @@ const MaybeRemoveSecurity: ResponseMiddleware = function () {
   this.next()
 }
 
-const GzipBody: ResponseMiddleware = function () {
+const GzipBody: ResponseMiddleware = async function () {
+  if (this.protocolManager && this.req.browserPreRequest?.requestId) {
+    const span = telemetry.startSpan({ name: 'gzip:body:protocol-notification', parentSpan: this.resMiddlewareSpan, isVerbose })
+    const resultingStream = this.protocolManager.responseStreamReceived({
+      requestId: this.req.browserPreRequest.requestId,
+      responseHeaders: this.incomingRes.headers,
+      isAlreadyGunzipped: this.isGunzipped,
+      responseStream: this.incomingResStream,
+      res: this.res,
+    })
+
+    if (resultingStream) {
+      this.incomingResStream = resultingStream.on('error', this.onError).once('finish', () => {
+        span?.end()
```

Comment:
if we don't get a resultingStream, we probably should end the span regardless to prevent telemetry issues and maybe add an attribute whether the resultingStream was found or not?

Reply:
Good catch: [`1148d29` (#27462)](https://github.com/cypress-io/cypress/pull/27462/commits/1148d29d6ef39a56c1db3ec643a48045e58bfe63)